vcd/resource_vcd_nsxt_route_advertisement.go

```diff
@@ -0,0 +1,184 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"log"
+	"strings"
+)
+
+func resourceVcdNsxtRouteAdvertisement() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtRouteAdvertisementCreateUpdate,
+		ReadContext:   resourceVcdNsxtRouteAdvertisementRead,
+		UpdateContext: resourceVcdNsxtRouteAdvertisementCreateUpdate,
+		DeleteContext: resourceVcdNsxtRouteAdvertisementDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtRouteAdvertisementImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "NSX-T Edge Gateway ID in which route advertisement is located",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Define if route advertisement is active",
+			},
+			"subnets": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Set of subnets that will be advertised to Tier-0 gateway. Leaving it empty means none",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceVcdNsxtRouteAdvertisementCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
```

Comment:
It is hard to get clarifications, but some resources require a lock on VDC Group when the Edge Gateway is a member of VDC Group. (https://github.com/vmware/terraform-provider-vcd/blob/main/vcd/resource_vcd_nsxt_ip_set.go#L80-L89)

Could you possibly tune-up test `TestAccVcdNsxVdcGroupCompleteMigration` to include route advertisement configuration as well, when the Edge Gateway is in VDC Group. Other option is to create some separate test, but we need to be sure that this interoperates well when the Edge Gateway is in VDC Group - I've been hit by that as well.

Reply:
@Didainius so I've added a conditional lock like other resources have so in case the NSX-T edge gateway is attached to a vdc group it will lock the group, and if that's not the case it'll lock the edge gateway otherwise.

Also please check that I've added a new test called `TestAccVcdNsxtRouteAdvertisementVdcGroup` that checks route advertisement fits good with vdc groups. I have tested just one case because it's exactly the same as testing it in a normal edge gateway, so we save some testing time.

Also this PR introduces new two fields `testConfig.Nsxt.VdcGroup` and `testConfig.Nsxt.VdcGroupEdgeGateway` to testConfig to use 😄  I'll update the PR description to reflect that.